modules/command/dv/dv-purchase-initiate-command.js

```diff
@@ -137,9 +138,10 @@ class DvPurchaseInitiateCommand extends Command {
         }, { where: { handler_id } });
     }
 
-    async _validatePrivateDataRootHash(dataSetId, otObjectId, private_data_root_hash) {
-        const privateObject = await this.importService.getPrivateDataObject(dataSetId, otObjectId);
-        return private_data_root_hash === privateObject.private_data_hash;
+    async _validatePermissionedDataRootHash(dataSetId, otObjectId, permissioned_data_root_hash) {
+        const permissionedDataObject =
+            await this.importService.getPermissionedDataObject(dataSetId, otObjectId);
```

Comment:
I would put this in separate service permission-data-service. I think that we should group methods depending on their functionality eg. Everything that is connected to import process should go to import-service, everything connected to permissioned-data should go in permission-data-service...